### DISCUSSION TITLE
GuardBreak

### DISCUSSION THREAD

**Discussion by ala-cpu** [ikemen-engine/Ikemen-GO#1801] [Q&A]
how to detected if GuardBreak on or off in game options

example

[State 0, ChangeState]
type = ChangeState
triggerall = GuardBreak = 1 : on in game options
trigger1=(stateno=150 || stateno=152 || stateno=154) && alive && !ishelper && sysfvar(0)>id && playeridexist(floor(sysfvar(0)))
trigger1=playerid(floor(sysfvar(0))),var(0)=90900 && playerid(floor(sysfvar(0))),var(36)>=50
value=190010
ignorehitpause=1

[State 0, ChangeState]
type = ChangeState
triggerall = !GuardBreak  : off in game options
trigger1=(stateno=150 || stateno=152 || stateno=154) && alive && !ishelper && sysfvar(0)>id && playeridexist(floor(sysfvar(0)))
trigger1=playerid(floor(sysfvar(0))),var(0)=90900 && playerid(floor(sysfvar(0))),var(36)>=50
value=190010
ignorehitpause=1

**Comment by potsmugen** [comment]
You can't currently. What I do is check if the guard points are full (after a move you *know* should take away some). But we should have triggers for game options, yeah.

**Reply by ala-cpu** [reply]
You are right, but I wanted to hide the guard bar from the beginning of the round, but after I checked the guard points in Debug, it became clear to me that it remains 1000, even if I close the guard break from options, so this is what I did.

[state 0]
	type=palfx
	triggerall = guardPoints <= 999.999
	trigger1=(stateno=150 || stateno=152 || stateno=154) && alive && !ishelper && (gametime%5=0) && sysfvar(0)>id && playeridexist(floor(sysfvar(0)))
	trigger1=playerid(floor(sysfvar(0))),var(0)=90900 && playerid(floor(sysfvar(0))),var(36)>=40
	time=2
	add=200,0,0

thank you for comment
